### PR TITLE
[K3.1]: Let the user set the item selected by default on recent discussions and search views #2489

### PR DESCRIPTION
**Pull request by 810**
merge this one after the #2777

### CODE REVIEW

File: components/com_kunena/site/controllers/user.php [hunk 1]
```diff
@@ -573,6 +573,7 @@ protected function saveSettings() {
 		$this->me->hideEmail = JRequest::getInt('hidemail', '');
 		$this->me->showOnline = JRequest::getInt('showonline', '');
 		$this->me->canSubscribe = JRequest::getInt('cansubscribe', '');
+		$this->me->userListtime = JRequest::getInt('userlisttime', '');
```
Comment (xillibit) [inline components/com_kunena/site/controllers/user.php]: You should use JInput isntead JRequest when modifing/adding some parts of code

**Comment by rich20** [comment]
### Backend
- [x] Selected time period for Search - Success

### Frontend
- [x] Profile:  Select time period for Recent Topics - Success
- [ ] Recent Topics: Temporary Time Period - Failed (The time periods are permanently defined. You can't change the time period temporarily at the Recent Topics page.)

**Comment by 810** [comment]
could you check if the model has changed for you:
components/com_kunena/site/models/topics.php 

Also check if the setting is saved

**Comment by rich20** [comment]
Maybe you misunderstood me. The profile setting works. But there is also a drop down menu at the Recent Topics page. And here you can't switch.

**Comment by 810** [comment]
ok, i thought you getting an error. But the switch doesn't work anymore, is always the user setting. @xillibit do you have any idea, to override this

**Comment by 810** [comment]
@rich20 i think i fix it, could you check it

**Comment by 810** [comment]
closing this one: see https://github.com/Kunena/Kunena-Forum/pull/2800

**Comment by rich20** [comment]
It's not optimal. If you switch on recent topic page it is only temporary. After page refresh it switch back to the user setting. The same if you go to the next page.

**Pull request closed without merging**